### PR TITLE
Some refactoring, channel and ordering of tasks.

### PR DESCRIPTION
**Pull request by pingpongsneak**

### CODE REVIEW

File: cypcore/Ledger/Graph.cs [hunk 1]
```diff
@@ -30,18 +30,17 @@ namespace CYPCore.Ledger
...
         Task<IEnumerable<Block>> GetSafeguardBlocks();
         Task<ulong> GetHeight();
         Task<BlockHash> GetHash(ulong height);
+        AsyncAgent<BlockGraph> BlockGraphAgent { get; }
     }
 
     public sealed class Graph : IGraph
     {
-        public const uint BlockmaniaTimeSlot = 0x0000007;
+        public const uint BlockmaniaTimeSlot = 0x000000A;
```
Comment (stephankempkes) [inline cypcore/Ledger/Graph.cs:43]: Is there a reason to define this in hex?
Maybe including the unit (secs) in the variable name gives some more clarify to what this variable really does.

File: cypcore/Ledger/Graph.cs [hunk 2]
```diff
@@ -400,122 +346,97 @@ private BlockGraph CopyBlockGraph(byte[] block, byte[] prevBlock)
...
         {
             Guard.Argument(blockGraph, nameof(blockGraph)).NotNull();
-            var tcs = new TaskCompletionSource<bool>();
-
-            async void TrySave()
+            try
             {
-                try
+                var copy = false;
+                copy |= blockGraph.Block.Node != _serfClient.ClientId;
```
Comment (stephankempkes) [inline cypcore/Ledger/Graph.cs:355]: Isn't this the same as "var copy = blockGraph.Block.Node != _serfClient.ClientId"?

File: cypcore/Ledger/Graph.cs [hunk 3]
```diff
@@ -400,122 +346,97 @@ private BlockGraph CopyBlockGraph(byte[] block, byte[] prevBlock)
...
+                switch (copy)
                 {
-                    var copy = false;
-                    copy |= blockGraph.Block.Node != _serfClient.ClientId;
-                    switch (copy)
-                    {
-                        case false:
+                    case false:
```
Comment (stephankempkes) [inline cypcore/Ledger/Graph.cs:358]: Why no if/else for boolean instead of switch?

File: cypcore/Ledger/PosMinting.cs [hunk 4]
```diff
@@ -75,122 +73,111 @@ public class PosMinting : IPosMinting
...
+        private async Task Stake()
         {
             if (_sync.SyncRunning) return;
             if (StakeRunning) return;
 
-            async void TryStake()
+            try
             {
-                try
+                if (_stakingConfigurationOptions.OnOff != true)
```
Comment (stephankempkes) [inline cypcore/Ledger/PosMinting.cs:100]: Just as a reminder to myself, we should rename this to "staking enabled". Staking OnOff == true is confusing.

Comment (sanastasiou) [reply cypcore/Ledger/PosMinting.cs:100]: Simply staking would do as well.

**Review by stephankempkes — APPROVED** [review]

File: cypcore/Ledger/PosMinting.cs (hunk 4 above)
Comment (stephankempkes) [reply cypcore/Ledger/PosMinting.cs:100]: It's a field in a section already named "Staking": https://github.com/cypher-network/cypher/blob/master/cypnode/Configuration/Templates/appsettings.json#L53

Comment (sanastasiou) [reply cypcore/Ledger/PosMinting.cs:100]: Then simply enabled would be just if not better.

**Review by sanastasiou — APPROVED** [review]

File: cypcore/Ledger/Graph.cs (hunk 1 above)
Comment (pingpongsneak) [reply cypcore/Ledger/Graph.cs:43]: Theres no other reason not to display the numbers in a base 16 numeral system. 
Defining/including the secs in the variable name will definitely change overtime.   As we are still tweaking the sweet spot of the time slot.  So it's better the have it as a constant.

**Pull request merged**